kernel/sysproc.c

```diff
@@ -6,6 +6,11 @@
 #include "mmu.h"
 #include "proc.h"
 
+extern struct {
+  struct spinlock lock;
+  struct proc proc[NPROC];
+} ptable;
+
```

Comment:
Moving the ptable definition of `ptable` from proc.h to sysproc.c should probably be in a separate commit.